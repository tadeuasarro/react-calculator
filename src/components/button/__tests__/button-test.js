import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '../button';

const handleClick = buttonName => {
  setState(state => Calculate(state, buttonName));
}

describe("Button", () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button name="0" onClick={handleClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a Button component', () => {
    render(<Button name="0" onClick={handleClick} />);
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument()
  });
});