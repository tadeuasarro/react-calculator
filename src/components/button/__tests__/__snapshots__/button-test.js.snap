// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="button-gray button-shape"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;
